### ISSUE TITLE
Abstract perform_https_request so that it can be used in Wasm context

### ISSUE THREAD

**Issue by carlopi** [evidence-dev/duckdb_gsheets#35]
This is something I touched on recently while looking at what would it take to make possible for https://github.com/duckdb/uc_catalog to work in a Web context.

Problem is basically how to wrap a minimal API that allows to make requests either natively OR from WebAssembly (there Web and Node might be two slightly different challenges).

I think I know what's the way forward, and implementing a curl wrapper (say as a header) would allow to solve the problems in:
* duckdb/uc_catalog
* evidence-dev/duckdb_gsheets
and possibly elsewhere.

I think it's likely on me, I can ping when I have a solution.

Given it's currently not functional, I opened a PR to disable Wasm, @archiewood, WDYT? (https://github.com/duckdb/community-extensions/pull/169)

**Comment by archiewood** [comment]
Agreed, I would love to bring this to WASM.

I'm not particularly happy with the implementation of `perform_http_request()` using OpenSSL as it is. It feels pretty dirty.

I was experimenting with [libcurl](https://curl.se/libcurl/c/) but couldn't get it to work

**Comment by pspeter3** [comment]
WASM support for GSheets would be awesome!

**Comment by kellington** [comment]
Agree.  Allowing SQL on a Google Sheet via [shell.](https://shell.duckdb.org) would be fantastic

**Comment by carlopi** [comment]
Hello!
I might have something to share, I have a fork of `duckdb_gsheets` that is compatible with the WasmHttpLib, I do need to rework a bit but will send a PR soon.

With that in, and disabling CORS, I get:
<img width="1730" height="346" alt="Image" src="https://github.com/user-attachments/assets/c9aa5e2e-61da-4127-9aec-bd4aaa283d6a" />

(I provided the token manually, possibly the token generation also could be adapted to be browser-aware, but that's just convenience).

Now the main problem: what would be a reasonable setup for a CORS-proxy? Or even better, are there alternatives to setting up a proxy for https://sheets.googleapis.com?

**Comment by archiewood** [comment]
Nice. 

As a side question, what formatter do you use. I'm such a C novice that I have no idea how this works.

In js i would just have a prettier.config.js file and `npm run format`, but is there an equivalent that I should add for C++?

When it was just me working on this code, formatting seemed not an issue, but now every PR has gnarly diffs

**Comment by archiewood** [comment]
> Now the main problem: what would be a reasonable setup for a CORS-proxy? Or even better, are there alternatives to setting up a proxy for https://sheets.googleapis.com/?

As I understand it this would only be an issue for POST requests to the google sheets API, so reading would work, but writing would be blocked without a CORS solution?

This _might_ be an acceptable outcome.

I suspect the challenge with a CORS proxy is who would host it. I don't think users would want to push requests through my server, and I wouldn't really want to run such a server.

**Comment by pspeter3** [comment]
For private sheets, do you need CORS? Google Drive / Sheets already has JS API support.
